cpp/include/Ice/Proxy.h

```diff
@@ -165,7 +165,20 @@ namespace Ice
          * @param timeout The new invocation timeout (in milliseconds).
          * @return A proxy with the new timeout.
          */
-        Prx ice_invocationTimeout(int timeout) const { return fromReference(asPrx()._invocationTimeout(timeout)); }
+        Prx ice_invocationTimeout(int timeout) const
+        {
+            return ice_invocationTimeout(std::chrono::milliseconds(timeout));
+        }
+
+        /**
+         * Obtains a proxy that is identical to this proxy, except for the invocation timeout.
+         * @param timeout The new invocation timeout (in milliseconds).
+         * @return A proxy with the new timeout.
+         */
+        Prx ice_invocationTimeout(std::chrono::milliseconds timeout) const
```

Comment:
What about using a `const std::chrono::duration<Rep, Period>&`, and let the user pick the unit for this duration?

See for example: https://en.cppreference.com/w/cpp/thread/future/wait_for

Reply:
How would we store the timeout?

Say you pass `chrono::microseconds(1)` to `ice_invocationTimeout` if we only store timeout as milliseconds, the conversion would produce `0ms`.

Note that you can still call `ice_invocationTimeout(1s)`, chrono::duration seems to automatically handled this loss less conversions.

Reply:
I would store it as milliseconds. 

> Say you pass chrono::microseconds(1) to ice_invocationTimeout if we only store timeout as milliseconds, the conversion would produce 0ms.

That's fine. The conversion (in the implementation) can be lossy.

Reply:
I updated the code to use a template duration parameter but because locator cache timeout is stored as seconds these two calls are equivalent:

```
ice_locatorCacheTimeout(100ms);  
ice_locatorCacheTimeout(0s);
```

I guess I should update the code to also store the locator cache timeout in milliseconds and don't worry about sub-milliseconds timeouts which doesn't make much sense.